plugins/inputs/couchdb/couchdb.go

```diff
@@ -166,6 +178,66 @@ func (c *CouchDB) fetchAndInsertData(accumulator telegraf.Accumulator, host stri
 	return nil
 }
 
+func (c *CouchDB) fetchAndInsertDbData(accumulator telegraf.Accumulator, host string) error {
+
+	println("Fetch db info for " + host)
+	response, error := client.Get(host)
+	if error != nil {
+		return error
+	}
+	defer response.Body.Close()
+
+	var dbs []string
+
+	decoder := json.NewDecoder(response.Body)
+	decoder.Decode(&dbs)
+
+	for _, db := range dbs {
+
+		fields := map[string]interface{}{}
+
+		response, error := client.Get(strings.Replace(host, "_all_dbs", db, -1))
+		if error != nil {
+			return error
+		}
+		defer response.Body.Close()
+		bodyBytes, _ := ioutil.ReadAll(response.Body)
+		var dat map[string]interface{}
```

Comment:
Can you define a new type for the response, it should look a bit like the Stats struct.

Reply:
I think this adds a lot of boiler plate code since the struct returned from the server is much larger than the data we can actually use (since it includes deprecated fields that I therefore do not parse). The content of the data can also be easily seen by following the documentation link I included... so did not change this.

Reply:
You should be able to only define the fields you would like to use.